libqtile/notify.py

```diff
@@ -213,19 +212,20 @@ def add(self, notif):
         with self._lock:
             # Find the id of the previous notification
             if self._notifications:
-                prev_id = next(reversed(self._notifications))
-                prev_id = self._notifications[prev_id].id
+                # Python 3.7 compatibility. For python 3.8 and later ommit
+                # the list since dict and dict_keys are reversible.
+                prev_id = next(reversed(list(self._notifications.keys())))
```

Comment:
> list-ing a dictionary already yields just the keys:
 Yeah you are right, I already knew that I wanted to put the keys to basically make it explicit that they key is the id. I will fix it.

> You can also simplify the next-reversed just by indexing:
When I wrote this i was between what you wrote and this. I just wanted to leave it like this because creating a list just to take the last element of a dict is not actually the most efficient thing (but required for Python 3.7). It is just there to indicate that in case Qtile changes minimum Python version requirement, list should be dropped and use next(reversed()) for efficiency. I could also make version checking but I think it is an overkill since notifications are not going to be coming bursting.

I will make the changes and update the comment.